client/my-sites/plans-grid/components/actions.tsx

```diff
@@ -306,15 +296,17 @@ const LoggedInPlansFeatureActionButton = ( {
 					{ translate( 'Upgrade' ) }
 				</Button>
 			);
+		} else if ( planActionOverrides?.currentPlan ) {
+			const { callback, text } = planActionOverrides.currentPlan;
+			return (
+				<Button className={ classes } disabled={ ! callback } onClick={ callback }>
+					{ text }
+				</Button>
+			);
 		}
-
 		return (
-			<Button
-				className={ classes }
-				href={ currentPlanManageHref }
-				disabled={ ! currentPlanManageHref }
-			>
-				{ canUserManageCurrentPlan ? translate( 'Manage plan' ) : translate( 'View plan' ) }
+			<Button className={ classes } disabled>
+				{ translate( 'Active Plan' ) }
```

Comment:
Thanks for doing this refactor. 

NIT / or to think about for later (if it makes sense): I wonder if a "general default state" would work better instead. So all CTAs to have same default state ("Contact support" perhaps) that is overriden either by the `planActionOverrides` or internally in actions as done already (so "Upgrade" or "Upgrade to ..", etc) are all replacing "Contact support". That may be the case already if we chase the code, but I think it could be more clear maybe.

Reply:
Yes ideally this would make the code much simpler. The only problem is with testing all the related variants. The action buttons are a giant bowl of spaghetti where changes have been done for various use cases which are technically not the concern of the plans grid.

Once we complete the plans grid packaging work it would be a good immediate follow up to audit all use cases. Cleanup the money dead case (I have a hunch there a few) and possibly force the client to inject the related text and possible styles and callback related to the button.

However I will see if a more clear "default" mechanism could be set in place to be overridden subsequently in a followup.